sdk/search/search-documents/review/search-documents.api.md

```diff
@@ -2618,23 +3024,29 @@ export class SearchIndexClient {
     readonly apiVersion: string;
     createAlias(alias: SearchIndexAlias, options?: CreateAliasOptions): Promise<SearchIndexAlias>;
     createIndex(index: SearchIndex, options?: CreateIndexOptions): Promise<SearchIndex>;
+    createKnowledgeAgent(knowledgeAgent: KnowledgeAgent, options?: CreateKnowledgeAgentOptions): Promise<KnowledgeAgent>;
     createOrUpdateAlias(alias: SearchIndexAlias, options?: CreateOrUpdateAliasOptions): Promise<SearchIndexAlias>;
     createOrUpdateIndex(index: SearchIndex, options?: CreateOrUpdateIndexOptions): Promise<SearchIndex>;
+    createOrUpdateKnowledgeAgent(agentName: string, knowledgeAgent: KnowledgeAgent, options?: CreateOrUpdateKnowledgeAgentOptions): Promise<KnowledgeAgent>;
     createOrUpdateSynonymMap(synonymMap: SynonymMap, options?: CreateOrUpdateSynonymMapOptions): Promise<SynonymMap>;
     createSynonymMap(synonymMap: SynonymMap, options?: CreateSynonymMapOptions): Promise<SynonymMap>;
     deleteAlias(alias: string | SearchIndexAlias, options?: DeleteAliasOptions): Promise<void>;
     deleteIndex(index: string | SearchIndex, options?: DeleteIndexOptions): Promise<void>;
+    deleteKnowledgeAgent(agent: string | KnowledgeAgent, options?: DeleteKnowledgeAgentOptions): Promise<void>;
```

Comment:
Is this either an agent name or a `KnowledgeAgent` instance?

Reply:
Yes, this is to align with the existing surface of `deleteIndex`.

Reply:
I think an overload would help here. 

```ts
    deleteKnowledgeAgent(agentName: string, options?: DeleteKnowledgeAgentOptions): Promise<void>;
    deleteKnowledgeAgent(agent: KnowledgeAgent, options?: DeleteKnowledgeAgentOptions): Promise<void>;
```

Otherwise the `string` version doesnt help me understand what is this a string of?

Reply:
Sounds good to me! Done. I also updated some of the other APIs to match.